### PR TITLE
ERC-7540: Asynchronous ERC-4626 Tokenized Vaults

### PR DESCRIPTION
**Pull request by Joeysantoro**
ported from https://github.com/ethereum/EIPs/pull/7876

### CODE REVIEW

File: ERCS/erc-7540.md [hunk 1]
```diff
@@ -0,0 +1,323 @@
...
+- Claimed: the state where a Request is finalized by the user and the user receives the output token (e.g. `shares` for a deposit Request)
+- Claim function: the corresponding Vault method to bring a Request to Claimed state (e.g. `deposit` or `mint` claims `shares` from `requestDeposit`). Lower case claim always describes the verb action of calling a Claim function.
+- operator: the account specified by the sender of the Request which has the right to claim a given Request once it is Claimable
+- asynchronous deposit Vault: a Vault that implements asynchronous Requests for deposit flows
+- asynchronous redemption Vault: a Vault that implements asynchronous redemption flows
+- fully asynchronous Vault: a Vault that implements asynchronous Requests for both deposit and redemption
+
+### Request Flows
+
+EIP-X Vaults MUST implement one or both of asynchronous deposit and redemption Request flows. If either flow is not implemented in a Request pattern, it MUST use the ERC-4626 standard synchronous interaction pattern.
```
Comment (lightclient) [inline ERCS/erc-7540.md:48]:
```suggestion
ERC-7540 Vaults MUST implement one or both of asynchronous deposit and redemption Request flows. If either flow is not implemented in a Request pattern, it MUST use the ERC-4626 standard synchronous interaction pattern. 
```

File: ERCS/erc-7540.md [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+- operator: the account specified by the sender of the Request which has the right to claim a given Request once it is Claimable
+- asynchronous deposit Vault: a Vault that implements asynchronous Requests for deposit flows
+- asynchronous redemption Vault: a Vault that implements asynchronous redemption flows
+- fully asynchronous Vault: a Vault that implements asynchronous Requests for both deposit and redemption
+
+### Request Flows
+
+EIP-X Vaults MUST implement one or both of asynchronous deposit and redemption Request flows. If either flow is not implemented in a Request pattern, it MUST use the ERC-4626 standard synchronous interaction pattern. 
+
+All EIP-X asynchronous tokenized Vaults MUST implement ERC-4626 with overrides for certain behavior described below.
```
Comment (lightclient) [inline ERCS/erc-7540.md:50]:
```suggestion
All ERC-7540 asynchronous tokenized Vaults MUST implement ERC-4626 with overrides for certain behavior described below.
```

**Review by lightclient — CHANGES_REQUESTED** [review]

File: ERCS/erc-7540.md (hunk 1 above)
Comment (lightclient) [reply ERCS/erc-7540.md:48]: @SamWilsn should eipw fail on this?

File: ERCS/erc-7540.md [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+- Claimed: the state where a Request is finalized by the user and the user receives the output token (e.g. `shares` for a deposit Request)
+- Claim function: the corresponding Vault method to bring a Request to Claimed state (e.g. `deposit` or `mint` claims `shares` from `requestDeposit`). Lower case claim always describes the verb action of calling a Claim function.
+- operator: the account specified by the sender of the Request which has the right to claim a given Request once it is Claimable
+- asynchronous deposit Vault: a Vault that implements asynchronous Requests for deposit flows
+- asynchronous redemption Vault: a Vault that implements asynchronous redemption flows
+- fully asynchronous Vault: a Vault that implements asynchronous Requests for both deposit and redemption
+
+### Request Flows
+
+ERC-7540 Vaults MUST implement one or both of asynchronous deposit and redemption Request flows. If either flow is not implemented in a Request pattern, it MUST use the ERC-4626 standard synchronous interaction pattern.
```
Comment (Joeysantoro) [inline ERCS/erc-7540.md:48]:
```suggestion
[ERC-7540](./erc-7540.md) Vaults MUST implement one or both of asynchronous deposit and redemption Request flows. If either flow is not implemented in a Request pattern, it MUST use the ERC-4626 standard synchronous interaction pattern. 
```

**Comment by Joeysantoro** [comment]
linking https://github.com/ethereum/ERCs/pull/10#discussion_r1373721970 @lightclient @SamWilsn I think EIP-X should probably fail eipw

**Comment by Joeysantoro** [comment]
> The commit [787983e](https://github.com/ethereum/ERCs/commit/787983e3e6e1f82a80d81d75d563a60212e1bfd4) (as a parent of [60895d8](https://github.com/ethereum/ERCs/commit/60895d866a79dc42d19c3d0ac6923809425fb865)) contains errors. Please inspect the [Run Summary](https://github.com/ethereum/EIPs/pull/10/files) for details.

@lightclient the Run Summary links EIP repo with a mismatched PR

**Review by lightclient — APPROVED** [review]

**Pull request merged**